File: Examples/TrenchDeposition/TrenchDeposition.cpp
#include <Geometries/psMakeTrench.hpp>
#include <SimpleDeposition.hpp>
#include <psProcess.hpp>
#include <psToSurfaceMesh.hpp>
#include <psVTKWriter.hpp>
#include <psWriteVisualizationMesh.hpp>

int main() {
  using NumericType = double;
  constexpr int D = 2;

  auto geometry = psSmartPointer<psDomain<NumericType, D>>::New();
  psMakeTrench<NumericType, D>(
      geometry, 0.02 /* grid delta */, 1.0 /*x extent*/, 1.0 /*y extent*/,
      0.4 /*trench width*/, 0.8 /*trench height*/, 0.0 /* tapering angle */)
      .apply();

  // copy top layer to capture deposition
  geometry->duplicateTopLevelSet();

  auto model = psSmartPointer<SimpleDeposition<NumericType, D>>::New(
      0.1 /*particle sticking probability*/, 1.0 /*particel source power*/);

  psProcess<NumericType, D> process;
  process.setDomain(geometry);
  process.setProcessModel(model);
  process.setNumberOfRaysPerPoint(1000);
  process.setProcessDuration(5.0);

  geometry->printSurface("initial.vtp");

  process.apply();

  geometry->printSurface("final.vtp");

  if constexpr (D == 2)
    psWriteVisualizationMesh<NumericType, D>(geometry, "final").apply();
}

File: Examples/TrenchDeposition/TrenchDeposition.py
"""
Example demonstrating simple deposition in a trench
"""
import viennaps2d as vps


def main() -> None:
    gridDelta = 0.02
    xExtent = 1.0
    yExtent = 1.0
    trenchWidth = 0.4
    trenchHeight = 0.8
    taperAngle = 0.0
    processTime = 5.0
    stickingProbability = 0.1
    sourceExponent = 1.0

    geometry = vps.psDomain()
    vps.psMakeTrench(geometry, gridDelta,
                     xExtent, yExtent,
                     trenchWidth, trenchHeight,
                     taperAngle).apply()

    model = vps.SimpleDeposition(
        stickingProbability=stickingProbability, sourceExponent=sourceExponent)

    process = vps.psProcess()
    process.setDomain(geometry)
    process.setProcessModel(model.getProcessModel())
    process.setNumberOfRaysPerPoint(1000)
    process.setProcessDuration(processTime)

    geometry.printSurface("initial.vtp")

    process.apply()

    geometry.printSurface("final.vtp")


if __name__ == "__main__":
    main()
